core/src/com/unciv/logic/map/mapunit/MapUnit.kt

```diff
@@ -509,6 +509,7 @@ class MapUnit : IsPartOfGameInfoSerialization {
         var healing = when {
             tile.isCityCenter() -> 25
             tile.isWater && isFriendlyTerritory && (baseUnit.isWaterUnit() || isTransported) -> 20 // Water unit on friendly water
+            tile.isWater && isFriendlyTerritory && (baseUnit.hasUnique(UniqueType.CanMoveOnWater)) -> 20 // Treated as a water unit on friendly water
```

Comment:
Because when I was working on it I didn't realize I was already in MapUnit, and didn't understand why it didn't let me use cache.CanMoveOnWater.
Didn't put it above for both line readability reasons and the comment